### PR TITLE
Upgrading and standardizing versions of Node

### PR DESCRIPTION
**Pull request by megoth**
Upgrading the default version we're using to build this project from `8.16` to `12.7`.

Also adding `8.10` as the lowest version we'll currently support (works with ESLint) so that we know when it breaks (and then fix it or reevaluate which versions we'll support).

`package-lock.json` is simply the new generated file after upgrading to `12.7`.

### CODE REVIEW

**Comment by megoth** [comment]
It should work with any versions newer than 8.10, so I think he's good, but yeah, I'll check with him ^_^

**Comment by Vinnl** [comment]
@megoth Yeah it should work, but with the newer version you'd also get a newer version of npm, and in between the lockfile format changed. So if Tim would continue to use the old version of npm, we'd get a new lockfile with many changes every time someone runs `npm install` with a different version than the last time the file was generated :)

**Comment by RubenVerborgh** [comment]
Tim needs to upgrade 😄

@megoth Is `DEFAULT_NODE_VERSION` still used anywhere? Looks like a leftover from older build scripts.

**Comment by megoth** [comment]
> @megoth Is `DEFAULT_NODE_VERSION` still used anywhere? Looks like a leftover from older build scripts.

I'll double-check, gotta admit it was shamelessly copied from https://github.com/rdfjs/N3.js/blob/master/.travis.yml#L3-L7 without much though -_-

**Comment by megoth** [comment]
I can't find any information for why it should be there, so I'll remove it.

**Comment by megoth** [comment]
@RubenVerborgh I was thinking of switching out `npm install` with `npm ci` ([npm-ci](https://docs.npmjs.com/cli/ci.html)), which means that we must test with Node v8.10.0 (npm v5.6.0) and Node v10.3.0 (nvm v6.1.0).

I think it would be good to have a clean install and be 100% sure that `package-lock.json` describes what dependencies to use.

Do you have thoughts on this?

**Comment by RubenVerborgh** [comment]
> Do you have thoughts on this?

`npm ci`.

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 node_js:
+  - "8.12"
   - "8"
-  - "10.0"
+  - "10.3"
   - "10"
   - "12.0"
   - "12"
+  - "lts/*"
   - "node"
```
Comment (timbl) [inline .travis.yml]: In https://github.com/solid/solid-panes/pull/204/files you have just `install: ` - you want  `before_install` here?

**Pull request merged**